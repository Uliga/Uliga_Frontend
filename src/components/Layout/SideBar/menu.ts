import PATH from "../../../constants/path";

const menu = [
  {
    title: "가계부",
    theme: "normal",
    iconName: "book",
    iconSize: "1.25rem",
    path: PATH.MAIN,
    subMenu: [
      {
        title: "작성",
        theme: "normal",
        iconName: "arrowRight",
        iconSize: "1.2rem",
        path: PATH.WRITE,
      },
      {
        title: "캘린더",
        theme: "normal",
        iconName: "arrowRight",
        iconSize: "1.2rem",
        path: PATH.MAIN,
      },
      {
        title: "내역",
        theme: "normal",
        iconName: "arrowRight",
        iconSize: "1.2rem",
        path: "",
      },
    ],
  },
  {
    title: "예산",
    theme: "normal",
    iconName: "pig",
    iconSize: "1.5rem",
    path: PATH.BUDGET,
  },
  {
    title: "분석",
    theme: "normal",
    iconName: "chart",
    iconSize: "1.25rem",
    path: "",
    subMenu: [
      {
        title: "주간별 · 월별",
        theme: "normal",
        iconName: "arrowRight",
        iconSize: "1.2rem",
        path: "",
      },
      {
        title: "카테고리별 분석",
        theme: "normal",
        iconName: "arrowRight",
        iconSize: "1.2rem",
        path: "",
      },
    ],
  },
  {
    title: "청약 · 대출 · 보험",
    theme: "normal",
    iconName: "bank",
    iconSize: "1.25rem",
    path: "",
  },
];

const bottomMenu = [
  {
    title: "내 정보",
    theme: "primary",
    path: "",
  },
  {
    title: "로그아웃",
    theme: "primary",
    path: "",
  },
];

export { menu, bottomMenu };
