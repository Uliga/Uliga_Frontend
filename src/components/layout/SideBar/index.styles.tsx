import styled from "styled-components";
import IconButton from "../../IconButton";
import Button from "../../Button";
import COLORS from "../../../constants/color";

const Container = styled.div`
  width: 17rem;
  height: 100%;
  background-color: white;
  align-items: center;
  border-left: 0.1rem solid ${COLORS.GREY[200]};
  border-right: 0.1rem solid ${COLORS.GREY[200]};
  position: fixed;
`;

const Top = styled.div`
  width: 100%;
  height: 4.6875rem;
  background-color: white;
  align-items: center;
  padding-left: 1rem;
  border-bottom: 0.1rem solid ${COLORS.GREY[200]};
`;
const Middle = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
`;

const Wrapper = styled.div`
  padding: 2rem 0 2rem 2rem;
  padding-left: 2rem;
  border-bottom: 0.1rem solid ${COLORS.GREY[200]};
`;
const Bottom = styled.div`
  padding-top: 0.5rem;
  width: 100%;
  background-color: white;
  align-items: center;
`;

const MainMenu = styled(IconButton)`
  font-weight: 700;
  justify-content: left;
  font-size: 1.4rem;
  &:hover {
    color: ${COLORS.BLUE};
  }
  color: ${COLORS.GREY[600]};
`;
const SubMenu = styled(IconButton)`
  width: 13rem;
  font-size: 1.25rem;
  padding-top: 1rem;
  justify-content: space-between;
  &:hover {
    color: ${COLORS.BLUE};
  }
  color: ${COLORS.GREY[500]};
`;

const BottomButton = styled(Button)`
  padding: 1.5rem 0 0 2rem;
  font-size: 1.3rem;
  background-color: transparent;
  color: ${COLORS.GREY[600]};
  &:hover {
    color: ${COLORS.BLUE};
    background-color: transparent;
  }
`;

export {
  Container,
  Top,
  Middle,
  Wrapper,
  Bottom,
  MainMenu,
  SubMenu,
  BottomButton,
};
