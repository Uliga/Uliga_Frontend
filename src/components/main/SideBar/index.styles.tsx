import React from "react";
import styled from "styled-components";
import * as colors from "../../../constants/color";
import IconButton from "../../common/IconButton";
import Button from "../../common/Button";

const Container = styled.div`
  width: 13.0625rem;
  height: 100%;
  background-color: white;
  align-items: center;
  border-left: 0.1rem solid ${colors.GREY[200]};
  border-right: 0.1rem solid ${colors.GREY[200]};
  position: fixed;
`;

const Menu1 = styled.div`
  width: 100%;
  height: 4.6875rem;
  background-color: white;
  align-items: center;
  border-bottom: 0.1rem solid ${colors.GREY[200]};
`;
const Menu2 = styled.div`
  width: 100%;
  background-color: white;
  align-items: center;
  border-bottom: 0.1rem solid ${colors.GREY[200]};
`;
const Menu2A = styled.div`
  padding-top: 0.5rem;
  width: 100%;
  background-color: white;
`;

const Menu3 = styled.div`
  width: 100%;
  height: 9.5rem;
  background-color: white;
  align-items: center;
  border-bottom: 0.1rem solid ${colors.GREY[200]};
`;
const Menu4 = styled.div`
  width: 100%;
  background-color: white;
  align-items: center;
  border-bottom: 0.1rem solid ${colors.GREY[200]};
`;
const Menu5 = styled.div`
  width: 100%;
  background-color: white;
  align-items: center;
`;

const HouseholdIconButton = styled(IconButton)`
  margin-top: 2.075rem;
  margin-bottom: 0.8rem;
  margin-left: 1rem;
  justify-content: left;
  font-size: 1.3rem;
  &:hover {
    color: ${colors.BLUE};
  }
`;
const PigIconButton = styled(IconButton)`
  margin-top: 1.4rem;
  margin-bottom: 2.075rem;
  font-size: 1.3rem;
  justify-content: left;
  margin-left: 1rem;
  &:hover {
    color: ${colors.BLUE};
  }
`;
const AnalIconButton = styled(IconButton)`
  margin-top: 2.075rem;
  margin-bottom: 0.65rem;
  margin-left: 1rem;
  font-size: 1.3rem;
  justify-content: left;
  &:hover {
    color: ${colors.BLUE};
  }
`;
const EtcServiceButton = styled(IconButton)`
  margin-top: 2.075rem;
  margin-bottom: 2.075rem;
  margin-left: 1rem;
  font-size: 1.3rem;
  justify-content: left;
  &:hover {
    color: ${colors.BLUE};
  }
`;
const Menu2aIconButton = styled(IconButton)`
  font-weight: 100;
  justify-content: space-between;
  padding-top: 0.4rem;
  padding-bottom: 0.4rem;
  margin-left: 1rem;
  &:hover {
    color: ${colors.BLUE};
  }
`;

const MyPageButton = styled(Button)`
  font-weight: 400;
  font-size: 1.3rem;
  margin-top: 2.075rem;
  margin-left: 0.5rem;
  background-color: transparent;
  color: BLACK;
  &:hover {
    color: ${colors.BLUE};
    background-color: transparent;
  }
`;
const LogoutButton = styled(Button)`
  font-weight: 400;
  font-size: 1.3rem;
  margin-left: 0.5rem;
  margin-top: 0.625rem;
  background-color: transparent;
  color: BLACK;
  &:hover {
    color: ${colors.BLUE};
    background-color: transparent;
  }
`;
export {
  Container,
  Menu1,
  Menu2,
  Menu2A,
  Menu3,
  Menu4,
  Menu5,
  HouseholdIconButton,
  PigIconButton,
  EtcServiceButton,
  AnalIconButton,
  Menu2aIconButton,
  MyPageButton,
  LogoutButton,
};
