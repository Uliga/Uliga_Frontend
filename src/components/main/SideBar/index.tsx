import React from "react";
import {
  Container,
  Menu1,
  Menu2,
  Menu2A,
  Menu3,
  Menu4,
  Menu5,
  HouseholdIconButton,
  PigIconButton,
  EtcServiceButton,
  AnalIconButton,
  Menu2aIconButton,
  MyPageButton,
  LogoutButton,
} from "./index.styles";

export default function SideBar() {
  return (
    <Container>
      <Menu1 />
      <Menu2>
        <HouseholdIconButton
          title="가계부"
          theme="normal"
          iconName="book"
          widthSize={7.0625}
          iconSize="1.25rem"
        />
        <Menu2A>
          <Menu2aIconButton
            reverseIconButton
            title="작성"
            theme="normal"
            iconName="arrowRight"
            widthSize={10}
            iconSize="1.2rem"
          />
          <Menu2aIconButton
            reverseIconButton
            title="캘린더"
            theme="normal"
            iconName="arrowRight"
            widthSize={10}
            iconSize="1.2rem"
          />{" "}
          <Menu2aIconButton
            reverseIconButton
            title="내역"
            theme="normal"
            iconName="arrowRight"
            widthSize={10}
            iconSize="1.25rem"
          />
        </Menu2A>
        <PigIconButton
          title="예산"
          theme="normal"
          iconName="pig"
          widthSize={5.6}
          iconSize="1.5rem"
        />
      </Menu2>
      <Menu3>
        {" "}
        <AnalIconButton
          title="분석"
          theme="normal"
          iconName="chart"
          widthSize={6.0625}
          iconSize="1.25rem"
        />
        <Menu2A>
          <Menu2aIconButton
            reverseIconButton
            title="주간별 · 월별"
            theme="normal"
            iconName="arrowRight"
            widthSize={10}
            iconSize="1.2rem"
          />
          <Menu2aIconButton
            reverseIconButton
            title="카테고리별 분석"
            theme="normal"
            iconName="arrowRight"
            widthSize={10}
            iconSize="1.2rem"
          />
        </Menu2A>
      </Menu3>

      <Menu4>
        <EtcServiceButton
          title="청약 · 대출 · 보험"
          theme="normal"
          iconName="bank"
          widthSize={11.0625}
          iconSize="1.25rem"
        />
      </Menu4>
      <Menu5>
        <MyPageButton title="내 정보" size="medium" theme="primary" />
        <LogoutButton title="로그아웃" size="medium" theme="primary" />
      </Menu5>
    </Container>
  );
}
