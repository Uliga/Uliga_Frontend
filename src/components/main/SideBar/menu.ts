const menu = [
  {
    title: "가계부",
    theme: "normal",
    iconName: "book",
    iconSize: "1.25rem",
    subMenu: [
      {
        title: "작성",
        theme: "normal",
        iconName: "arrowRight",
        iconSize: "1.2rem",
      },
      {
        title: "캘린더",
        theme: "normal",
        iconName: "arrowRight",
        iconSize: "1.2rem",
      },
      {
        title: "내역",
        theme: "normal",
        iconName: "arrowRight",
        iconSize: "1.2rem",
      },
    ],
  },
  {
    title: "예산",
    theme: "normal",
    iconName: "pig",
    iconSize: "1.5rem",
  },
  {
    title: "분석",
    theme: "normal",
    iconName: "chart",
    iconSize: "1.25rem",
    subMenu: [
      {
        title: "주간별 · 월별",
        theme: "normal",
        iconName: "arrowRight",
        iconSize: "1.2rem",
      },
      {
        title: "카테고리별 분석",
        theme: "normal",
        iconName: "arrowRight",
        iconSize: "1.2rem",
      },
    ],
  },
  {
    title: "청약 · 대출 · 보험",
    theme: "normal",
    iconName: "bank",
    iconSize: "1.25rem",
  },
];

const bottomMenu = [
  {
    title: "내 정보",
    theme: "primary",
  },
  {
    title: "로그아웃",
    theme: "primary",
  },
];

export { menu, bottomMenu };
