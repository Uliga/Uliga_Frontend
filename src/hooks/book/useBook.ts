import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
import { useNavigate } from "react-router-dom";
import {
  answerInvitation,
  loadBookInfo,
  loadBookList,
  loadCategory,
  loadBookMember,
  loadMonthItems,
  loadSchedule,
} from "../../api/book";
import { IBookInfo, IBookList } from "../../interfaces/book";
import { IItem } from "../../interfaces/item";
import QUERYKEYS from "../../constants/querykey";
import toastMsg from "../../components/Toast";

export default function useBook() {
  const navigate = useNavigate();

  const useSelectedBook = (bookid: any) => {
    const queryFn = () => loadBookInfo(+bookid);
    const { isLoading, error, data } = useQuery<IBookInfo>(
      [QUERYKEYS.LOAD_BOOK_INFO],
      queryFn,
    );
    return { isLoading, error, data };
  };

  const useBookList = () => {
    const { data } = useQuery<IBookList>(
      [QUERYKEYS.LOAD_BOOK_LIST],
      loadBookList,
    );
    return { data };
  };

  const useReplaceBook = (path: string, bookId: number) => {
    navigate(`${path}/${bookId}`);
    window.location.reload();
  };

  const queryClient = useQueryClient();

  const mutateInvitation = useMutation(["answerInvitation"], answerInvitation, {
    onSuccess: data => {
      toastMsg(data.join ? "초대 수락 성공" : "초대 거절 성공");
      queryClient.invalidateQueries([QUERYKEYS.LOAD_ME]);
    },
    onError: ({
      response: {
        data: { errorCode, message },
      },
    }) => {
      toastMsg(`${errorCode} / ${message}`);
    },
  });

  const useLoadMonthItems = (id: number, year: number, month: number) => {
    const date = `${year}/${month}`;
    const queryFn = () => loadMonthItems(id, date);
    const { data, refetch } = useQuery<IItem>(
      [QUERYKEYS.LOAD_MONTH_ITEM],
      queryFn,
    );
    return { data, refetch };
  };

  const useCategoryList = (id: number) => {
    const queryFn = () => loadCategory(id);
    const { data } = useQuery([QUERYKEYS.LOAD_CATEGORY], queryFn);
    const list = data?.categories;
    return list;
  };

  const useSchedule = () => {
    const { data } = useQuery([QUERYKEYS.LOAD_SCHEDULE], loadSchedule);
    return data;
  };

  return {
    useSelectedBook,
    useReplaceBook,
    useBookList,
    mutateInvitation,
    useLoadMonthItems,
    loadBookMember,
    useCategoryList,
    useSchedule,
  };
}
