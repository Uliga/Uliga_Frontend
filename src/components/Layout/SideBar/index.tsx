import React from "react";
import { useQuery } from "@tanstack/react-query";
import { useLocation, useNavigate, useParams } from "react-router-dom";
import useBook from "../../../hooks/book/useBook";
import {
  Container,
  Top,
  Middle,
  Wrapper,
  Bottom,
  MainMenu,
  SubMenu,
  BottomButton,
  BookName,
  BookInfo,
} from "./index.styles";
import { menu, bottomMenu } from "./menu";
import Person from "../../../assets/person";
import { IUserInfo } from "../../../interfaces/user";
import QUERYKEYS from "../../../constants/querykey";
import { loadMe, logout } from "../../../api/user";
import { AVATAR_COLORS } from "../../../constants/color";
import toastMsg from "../../Toast";
import PATH from "../../../constants/path";

export default function SideBar() {
  const { bookId } = useParams();
  const { useSelectedBook } = useBook();
  const { data } = useSelectedBook(Number(bookId));
  const navigate = useNavigate();
  const { data: me } = useQuery<IUserInfo | undefined>(
    [QUERYKEYS.LOAD_ME],
    loadMe,
  );
  const userAvatar = data?.members?.find(
    ele => ele.id === me?.memberInfo?.id,
  )?.avatarUrl;
  if (!data) return null;

  const location = useLocation();
  const currentPath = location.pathname.split("/")[1];

  console.log(location.pathname);
  const authLogout = async () => {
    try {
      await logout();
      localStorage.clear();
      toastMsg("로그아웃 되었습니다.");
      navigate("/");
    } catch (err) {
      console.log(err);
    }
  };
  return (
    <Container>
      <Top>
        <Person color={AVATAR_COLORS[userAvatar || ""]?.color} />
        <div>
          <BookName>{data.info.accountBookName}</BookName>
          <BookInfo>
            {data.numberOfMember.count}명의 멤버 ({data.info.relationShip})
          </BookInfo>
        </div>
      </Top>
      <Middle>
        {menu.map(ele => {
          return (
            <Wrapper key={ele.title}>
              <MainMenu
                title={ele.title}
                theme={ele.theme}
                iconName={ele.iconName}
                iconSize={ele.iconSize}
                isActive={
                  ele.path === PATH.BUDGET && ele.path === `/${currentPath}`
                }
                onClick={() => {
                  navigate(`${ele.path}/${bookId}`);
                }}
              />
              {ele.subMenu?.map(sub => {
                return (
                  <SubMenu
                    key={sub.title}
                    reverseIconButton
                    title={sub.title}
                    theme={sub.theme}
                    iconName={sub.iconName}
                    iconSize={sub.iconSize}
                    isActive={sub.path === `/${currentPath}`}
                    onClick={() => {
                      navigate(`${sub.path}/${bookId}`);
                    }}
                  />
                );
              })}
            </Wrapper>
          );
        })}
      </Middle>
      <Bottom>
        {bottomMenu.map(ele => {
          return (
            <BottomButton
              key={ele.title}
              title={ele.title}
              theme={ele.theme}
              isActive={
                ele.path === `${PATH.SETTING}${PATH.ME}` &&
                ele.path.split("/")[1] === currentPath
              }
              onClick={() => {
                navigate(`${ele.path}/${bookId}`);
                if (ele.title === "로그아웃") {
                  authLogout();
                }
              }}
            />
          );
        })}
      </Bottom>
    </Container>
  );
}
