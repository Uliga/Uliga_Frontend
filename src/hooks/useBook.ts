import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
import { useNavigate } from "react-router-dom";
import {
  answerInvitation,
  loadBookInfo,
  loadBookList,
  loadMonthItems,
} from "../api/book";
import { IBookInfo, IBookList } from "../interfaces/book";
import { IItem } from "../interfaces/item";
import QUERYKEYS from "../constants/querykey";
import PATH from "../constants/path";
import toastMsg from "../components/Toast";

export default function useBook() {
  const navigate = useNavigate();

  const useSelectedBook = (bookid: any) => {
    const queryFn = () => loadBookInfo(+bookid);
    const { isLoading, error, data } = useQuery<IBookInfo>(
      [QUERYKEYS.LOAD_BOOK_INFO],
      queryFn,
    );
    return { isLoading, error, data };
  };

  const useBookList = () => {
    const { data } = useQuery<IBookList>(
      [QUERYKEYS.LOAD_BOOK_LIST],
      loadBookList,
    );
    return { data };
  };

  const useReplaceBook = (bookId: number) => {
    navigate(`${PATH.MAIN}/${bookId}`);
    window.location.reload();
  };

  const queryClient = useQueryClient();

  const mutateInvitation = useMutation(["answerInvitation"], answerInvitation, {
    onSuccess: data => {
      toastMsg(data.join ? "초대 수락 성공" : "초대 거절 성공");
      queryClient.invalidateQueries([QUERYKEYS.LOAD_ME]);
    },
    onError: ({
      response: {
        data: { errorCode, message },
      },
    }) => {
      toastMsg(`${errorCode} / ${message}`);
    },
  });

  const useLoadMonthItems = (id: number, year: number, month: number) => {
    const date = `${year}/${month}`;
    const queryFn = () => loadMonthItems(id, date);
    const { data, refetch } = useQuery<IItem>(
      [QUERYKEYS.LOAD_MONTH_ITEM],
      queryFn,
    );
    return { data, refetch };
  };

  return {
    useSelectedBook,
    useReplaceBook,
    useBookList,
    mutateInvitation,
    useLoadMonthItems,
  };
}
