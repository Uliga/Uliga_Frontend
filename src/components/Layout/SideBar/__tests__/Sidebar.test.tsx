import React from "react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { BrowserRouter as Router } from "react-router-dom";
import { render } from "@testing-library/react";
import useBook from "../../../../hooks/book/useBook";
import SideBar from "../index";
import "@testing-library/jest-dom";

jest.mock("../../../../hooks/book/useBook");

const mockedUseSelectedBook = jest.fn(() => ({
  isLoading: false,
  error: null,
  data: {
    info: {
      accountBookAuthority: "USER",
      accountBookId: 1,
      accountBookName: "TEST",
      getNotification: false,
      isPrivate: true,
      relationShip: "TEST",
      avatarUrl: "YELLOW",
    },
    categories: [],
    members: [
      {
        id: 1,
        username: "TEST",
        accountBookAuthority: "ADMIN",
        avatarUrl: "YELLOW",
        email: "TEST",
      },
    ],
    numberOfMember: {
      count: 1,
    },
  },
}));

describe("<SideBar />", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  jest.mock("react-router-dom", () => ({
    useNavigate: () => jest.fn(),
    useLocation: () => jest.fn(),
    useParams: () => jest.fn(),
  }));

  it("사이드바 스냅샷 테스트", () => {
    (useBook as jest.Mock).mockReturnValue({
      useSelectedBook: mockedUseSelectedBook,
    });
    const testUtils = render(
      <QueryClientProvider client={new QueryClient()}>
        <Router>
          <SideBar onClickSideBar={jest.fn()} />
        </Router>
      </QueryClientProvider>,
    );
    expect(testUtils).toMatchSnapshot();
  });

  it("사이드바에 모든 메뉴가 있는지 확인한다.", async () => {
    (useBook as jest.Mock).mockReturnValue({
      useSelectedBook: mockedUseSelectedBook,
    });

    const testUtils = render(
      <QueryClientProvider client={new QueryClient()}>
        <Router>
          <SideBar onClickSideBar={jest.fn()} />
        </Router>
      </QueryClientProvider>,
    );

    testUtils.getByText("가계부");
    testUtils.getByText("작성");
    testUtils.getByText("내역");
    testUtils.getByText("금융 일정");
    testUtils.getByText("예산");
    testUtils.getByText("분석");
    testUtils.getByText("주간별 · 월별 분석");
    testUtils.getByText("카테고리별 분석");
    testUtils.getByText("청약 · 대출 · 보험");
    testUtils.getByText("설정");
    testUtils.getByText("로그아웃");
  });
});
