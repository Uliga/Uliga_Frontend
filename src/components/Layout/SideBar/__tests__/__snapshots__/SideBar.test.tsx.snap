// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> 사이드바 스냅샷 테스트 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hLBbgP btgaLm"
      >
        <div
          class="sc-eDvSVe ioEtvE"
        >
          <div
            class="sc-csuSiG grwbTt"
          >
            <svg
              fill="none"
              height="30"
              viewBox="0 2 15 24"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="6.8983"
                cy="3.89831"
                fill="#FFC144"
                r="3.89831"
              />
              <path
                clip-rule="evenodd"
                d="M7.18202 10.6158C7.15158 10.6158 7.12126 10.616 7.09106 10.6163C7.06087 10.616 7.03055 10.6158 7.0001 10.6158V10.6177C0.820362 10.7458 -0.222249 17.4055 0.035133 20.7514C0.035133 23.4834 3.24864 24.0652 5.717 23.994L5.71759 23.9998H9.09612L9.09516 23.9973C11.4465 23.9533 14.147 23.2558 14.147 20.7514C14.4044 17.4055 13.3618 10.7458 7.18202 10.6177V10.6158Z"
                fill="#FFC144"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <div>
            <div
              class="sc-kDvujY bBQcxY"
            >
              TEST
            </div>
            <div
              class="sc-ipEyDJ kXVDcG"
            >
              1
              명의 멤버 (
              TEST
              )
            </div>
          </div>
        </div>
        <div
          class="sc-jSUZER fDxcKE"
        >
          <div
            class="sc-gKPRtg bCZmMO"
          >
            <button
              class="sc-bcXHqe eiSaub sc-ftTHYK DFCqO"
              data-cy=""
            >
              <i
                class="sc-gswNZR hEimFU bi bi-journals"
                style="font-size: 1.25rem;"
              />
              가계부
            </button>
            <button
              class="sc-bcXHqe eiSaub sc-pyfCe ftmXUj"
              data-cy=""
            >
              작성
              <i
                class="sc-gswNZR hEimFU bi bi-chevron-right"
                style="font-size: 1.2rem;"
              />
            </button>
            <button
              class="sc-bcXHqe eiSaub sc-pyfCe ftmXUj"
              data-cy=""
            >
              캘린더
              <i
                class="sc-gswNZR hEimFU bi bi-chevron-right"
                style="font-size: 1.2rem;"
              />
            </button>
            <button
              class="sc-bcXHqe eiSaub sc-pyfCe ftmXUj"
              data-cy=""
            >
              내역
              <i
                class="sc-gswNZR hEimFU bi bi-chevron-right"
                style="font-size: 1.2rem;"
              />
            </button>
          </div>
          <div
            class="sc-gKPRtg bCZmMO"
          >
            <button
              class="sc-bcXHqe eiSaub sc-ftTHYK DFCqO"
              data-cy=""
            >
              <i
                class="sc-gswNZR hEimFU bi bi-calendar-event"
                style="font-size: 1.4rem;"
              />
              금융 일정
            </button>
          </div>
          <div
            class="sc-gKPRtg bCZmMO"
          >
            <button
              class="sc-bcXHqe eiSaub sc-ftTHYK DFCqO"
              data-cy=""
            >
              <i
                class="sc-gswNZR hEimFU bi bi-piggy-bank"
                style="font-size: 1.45rem;"
              />
              예산
            </button>
          </div>
          <div
            class="sc-gKPRtg bCZmMO"
          >
            <button
              class="sc-bcXHqe eiSaub sc-ftTHYK DFCqO"
              data-cy=""
            >
              <i
                class="sc-gswNZR hEimFU bi bi-clipboard-data"
                style="font-size: 1.25rem;"
              />
              분석
            </button>
            <button
              class="sc-bcXHqe eiSaub sc-pyfCe ftmXUj"
              data-cy=""
            >
              주간별 · 월별 분석
              <i
                class="sc-gswNZR hEimFU bi bi-chevron-right"
                style="font-size: 1.2rem;"
              />
            </button>
            <button
              class="sc-bcXHqe eiSaub sc-pyfCe ftmXUj"
              data-cy=""
            >
              카테고리별 분석
              <i
                class="sc-gswNZR hEimFU bi bi-chevron-right"
                style="font-size: 1.2rem;"
              />
            </button>
          </div>
          <div
            class="sc-gKPRtg bCZmMO"
          >
            <button
              class="sc-bcXHqe eiSaub sc-ftTHYK DFCqO"
              data-cy=""
            >
              <i
                class="sc-gswNZR hEimFU bi bi-bank"
                style="font-size: 1.25rem;"
              />
              청약 · 대출 · 보험
            </button>
          </div>
        </div>
        <div
          class="sc-iBYQkv iUvwJW"
        >
          <button
            class="sc-dkrFOg euTfju sc-jrcTuL qWRmF"
            data-cy=""
            type="button"
          >
            설정
          </button>
          <button
            class="sc-dkrFOg euTfju sc-jrcTuL qWRmF"
            data-cy=""
            type="button"
          >
            로그아웃
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hLBbgP btgaLm"
    >
      <div
        class="sc-eDvSVe ioEtvE"
      >
        <div
          class="sc-csuSiG grwbTt"
        >
          <svg
            fill="none"
            height="30"
            viewBox="0 2 15 24"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="6.8983"
              cy="3.89831"
              fill="#FFC144"
              r="3.89831"
            />
            <path
              clip-rule="evenodd"
              d="M7.18202 10.6158C7.15158 10.6158 7.12126 10.616 7.09106 10.6163C7.06087 10.616 7.03055 10.6158 7.0001 10.6158V10.6177C0.820362 10.7458 -0.222249 17.4055 0.035133 20.7514C0.035133 23.4834 3.24864 24.0652 5.717 23.994L5.71759 23.9998H9.09612L9.09516 23.9973C11.4465 23.9533 14.147 23.2558 14.147 20.7514C14.4044 17.4055 13.3618 10.7458 7.18202 10.6177V10.6158Z"
              fill="#FFC144"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div>
          <div
            class="sc-kDvujY bBQcxY"
          >
            TEST
          </div>
          <div
            class="sc-ipEyDJ kXVDcG"
          >
            1
            명의 멤버 (
            TEST
            )
          </div>
        </div>
      </div>
      <div
        class="sc-jSUZER fDxcKE"
      >
        <div
          class="sc-gKPRtg bCZmMO"
        >
          <button
            class="sc-bcXHqe eiSaub sc-ftTHYK DFCqO"
            data-cy=""
          >
            <i
              class="sc-gswNZR hEimFU bi bi-journals"
              style="font-size: 1.25rem;"
            />
            가계부
          </button>
          <button
            class="sc-bcXHqe eiSaub sc-pyfCe ftmXUj"
            data-cy=""
          >
            작성
            <i
              class="sc-gswNZR hEimFU bi bi-chevron-right"
              style="font-size: 1.2rem;"
            />
          </button>
          <button
            class="sc-bcXHqe eiSaub sc-pyfCe ftmXUj"
            data-cy=""
          >
            캘린더
            <i
              class="sc-gswNZR hEimFU bi bi-chevron-right"
              style="font-size: 1.2rem;"
            />
          </button>
          <button
            class="sc-bcXHqe eiSaub sc-pyfCe ftmXUj"
            data-cy=""
          >
            내역
            <i
              class="sc-gswNZR hEimFU bi bi-chevron-right"
              style="font-size: 1.2rem;"
            />
          </button>
        </div>
        <div
          class="sc-gKPRtg bCZmMO"
        >
          <button
            class="sc-bcXHqe eiSaub sc-ftTHYK DFCqO"
            data-cy=""
          >
            <i
              class="sc-gswNZR hEimFU bi bi-calendar-event"
              style="font-size: 1.4rem;"
            />
            금융 일정
          </button>
        </div>
        <div
          class="sc-gKPRtg bCZmMO"
        >
          <button
            class="sc-bcXHqe eiSaub sc-ftTHYK DFCqO"
            data-cy=""
          >
            <i
              class="sc-gswNZR hEimFU bi bi-piggy-bank"
              style="font-size: 1.45rem;"
            />
            예산
          </button>
        </div>
        <div
          class="sc-gKPRtg bCZmMO"
        >
          <button
            class="sc-bcXHqe eiSaub sc-ftTHYK DFCqO"
            data-cy=""
          >
            <i
              class="sc-gswNZR hEimFU bi bi-clipboard-data"
              style="font-size: 1.25rem;"
            />
            분석
          </button>
          <button
            class="sc-bcXHqe eiSaub sc-pyfCe ftmXUj"
            data-cy=""
          >
            주간별 · 월별 분석
            <i
              class="sc-gswNZR hEimFU bi bi-chevron-right"
              style="font-size: 1.2rem;"
            />
          </button>
          <button
            class="sc-bcXHqe eiSaub sc-pyfCe ftmXUj"
            data-cy=""
          >
            카테고리별 분석
            <i
              class="sc-gswNZR hEimFU bi bi-chevron-right"
              style="font-size: 1.2rem;"
            />
          </button>
        </div>
        <div
          class="sc-gKPRtg bCZmMO"
        >
          <button
            class="sc-bcXHqe eiSaub sc-ftTHYK DFCqO"
            data-cy=""
          >
            <i
              class="sc-gswNZR hEimFU bi bi-bank"
              style="font-size: 1.25rem;"
            />
            청약 · 대출 · 보험
          </button>
        </div>
      </div>
      <div
        class="sc-iBYQkv iUvwJW"
      >
        <button
          class="sc-dkrFOg euTfju sc-jrcTuL qWRmF"
          data-cy=""
          type="button"
        >
          설정
        </button>
        <button
          class="sc-dkrFOg euTfju sc-jrcTuL qWRmF"
          data-cy=""
          type="button"
        >
          로그아웃
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
